accord-core/src/test/java/accord/burn/BurnTest.java

```diff
@@ -230,37 +236,50 @@ static void burn(RandomSource random, TopologyFactory topologyFactory, List<Id>
 
         // not used for atomicity, just for encapsulation
         AtomicReference<Runnable> onSubmitted = new AtomicReference<>();
-        Consumer<Packet> responseSink = packet -> {
-            ListResult reply = (ListResult) packet.message;
-            if (replies[(int)packet.replyId] != null)
-                return;
-
+        Runnable maybeTriggerNextRequest = () -> {
             if (requestIndex.get() < requests.length)
             {
                 int i = requestIndex.getAndIncrement();
                 starts[i] = clock.incrementAndGet();
-                queue.add(requests[i]);
+                // TODO (review): why did i add this again?
+                queue.addNoDelay(requests[i]);
```

Comment:
there was a reason I added this, but its been too long I forgot... I can always revert and add back w/e the reason is detected again...

Reply:
I'm sure this is fine